### PR TITLE
feat: Windows Support

### LINKED ISSUE #70: Escape paths before matching with lua patterns
As seen in #67, paths containing % and other symbols can cause issues with lua patterns - we should escape or validate paths before matching against them.

### PR DESCRIPTION
**Pull request by tris203**
This removes the dependency on mini.test
Add windows support by changing the searched dirs to be ```vim.fn.stdpath("config")```
And adds tests for all  OS's to CI

Closes #70

### CODE REVIEW

File: lua/hawtkeys/ts.lua [hunk 1]
```diff
@@ -469,17 +464,23 @@ function M.get_all_keymaps()
             end
         end
     else
-        local paths = get_runtime_path()
+        local paths = vim.fn.stdpath("config")
```
Comment (willothy) [inline lua/hawtkeys/ts.lua]: This should always be a single path as far as I know

File: tests/hawtkeys/ui_spec.lua [hunk 2]
```diff
@@ -1,63 +1,64 @@
...
+                    local ui = require("hawtkeys.ui")
+                    return {
+                        ui.SearchBuf,
+                        ui.SearchWin,
+                        ui.ResultBuf,
+                        ui.ResultWin,
+                        ui.Namespace,
+                        ui.prompt_extmark,
+                    }
+                end)())
```
Comment (willothy) [inline tests/hawtkeys/ui_spec.lua]:
```suggestion
                ui.SearchBuf,
                ui.SearchWin,
                ui.ResultBuf,
                ui.ResultWin,
                ui.Namespace,
                ui.prompt_extmark
```

File: tests/hawtkeys/ui_spec.lua [hunk 3]
```diff
@@ -1,63 +1,64 @@
...
     end)
 
     describe("search", function()
         before_each(function()
-            child.lua([[require("hawtkeys.ui").show()]])
+            require("plenary.reload").reload_module("hawtkeys")
+            vim.cmd([[bufdo! bwipeout]])
+            require("hawtkeys").setup({})
+            vim.cmd([[lua require("hawtkeys.ui").show()]])
             SearchBuf, SearchWin, ResultBuf, ResultWin, Namespace, prompt_extmark =
```
Comment (willothy) [inline tests/hawtkeys/ui_spec.lua:22]:
```suggestion
            local ui = require("hawtkeys.ui")
            SearchBuf, SearchWin, ResultBuf, ResultWin, Namespace, prompt_extmark =
```

File: tests/hawtkeys/ui_spec.lua [hunk 4]
```diff
@@ -70,50 +71,54 @@ describe("ui", function()
...
+            vim.cmd([[bufdo! bwipeout]])
+            require("hawtkeys").setup({})
+            vim.cmd([[lua require("hawtkeys.ui").show_all()]])
+            ResultBuf, ResultWin, Namespace = unpack((function()
                 local ui = require("hawtkeys.ui")
                 return {
                     ui.ResultBuf,
                     ui.ResultWin,
                     ui.Namespace,
                 }
```
Comment (willothy) [inline tests/hawtkeys/ui_spec.lua]:
```suggestion
            local ui = require("hawtkeys.ui")
            ResultBuf, ResultWin, Namespace = ui.ResultBuf, ui.ResultWin, ui.Namespace
```

File: tests/hawtkeys/ui_spec.lua [hunk 5]
```diff
@@ -70,50 +71,54 @@ describe("ui", function()
...
-            ]]))
+            end)())
```
Comment (willothy) [inline tests/hawtkeys/ui_spec.lua]:
```suggestion
```

File: tests/hawtkeys/ui_spec.lua [hunk 6]
```diff
@@ -70,50 +71,54 @@ describe("ui", function()
...
-            child.lua([[require("hawtkeys.ui").show_dupes()]])
-            ResultBuf, ResultWin, Namespace = unpack(child.lua([[
+            vim.cmd([[lua require("hawtkeys.ui").show_dupes()]])
+            ResultBuf, ResultWin, Namespace = unpack((function()
                 local ui = require("hawtkeys.ui")
                 return {
                     ui.ResultBuf,
                     ui.ResultWin,
                     ui.Namespace,
                 }
```
Comment (willothy) [inline tests/hawtkeys/ui_spec.lua]:
```suggestion
            vim.cmd([[lua require("hawtkeys.ui").show_dupes()]])
            local ui = require("hawtkeys.ui")
            ResultBuf, ResultWin, Namespace = ui.ResultBuf, ui.ResultWin, ui.Namespace
```

File: tests/hawtkeys/ui_spec.lua [hunk 7]
```diff
@@ -70,50 +71,54 @@ describe("ui", function()
...
-            ]]))
+            end)())
```
Comment (willothy) [inline tests/hawtkeys/ui_spec.lua]:
```suggestion
```

**Review by willothy — CHANGES_REQUESTED** [review]
Just a few small things then lgtm

**Comment by tris203** [comment]
@willothy do you think this is ready to merge too?

**Comment by willothy** [comment]
> @willothy do you think this is ready to merge too?

CI passing so yessir, lgtm!

**Review by willothy — APPROVED** [review]

**Pull request merged**